src/theory/quantifiers/quant_conflict_find.cpp

```diff
@@ -615,27 +620,6 @@ bool QuantInfo::isTConstraintSpurious( QuantConflictFind * p, std::vector< Node
   return p->d_quantEngine->inConflict();
 }
 
-bool QuantInfo::isPropagatingInstance( QuantConflictFind * p, Node n ) {
-  if( n.getKind()==FORALL ){
-    //TODO?
-    return true;
-  }else if( n.getKind()==NOT || n.getKind()==AND || n.getKind()==OR || n.getKind()==EQUAL || n.getKind()==ITE || 
-            ( n.getKind()==EQUAL && n[0].getType().isBoolean() ) ){
-    for( unsigned i=0; i<n.getNumChildren(); i++ ){
-      if( !isPropagatingInstance( p, n[i] ) ){
-        return false;
-      }
-    }
-    return true;
-  }else{
-    if( p->getEqualityEngine()->hasTerm( n ) || isGroundSubterm( n ) ){
```

Comment:
removing the test of `isGroundSubterm( n )` (and the auxiliary information to set and test this) is what leads to the performance losses, isn't it? It does not seem to simplify things too much. I think we should keep this for now and only remove it when we add an alternative (and cleaner) way of dealing with these cases.

Reply:
There are a handful of sledgehammer examples that have axioms forall x : U. x = f where f is a ground term. It was by chance that this hack led to solving these benchmarks. In general, this is a bad idea.

The proper way to handle this is to fix our trigger selection to make x a "variable trigger" in this case.

Reply:
Why is it a bad idea in general?

Reply:
It's the same reason why we don't register terms that occur in quantifier bodies for E-matching.

Reply:
You tested and confirmed this is a bad idea? I had an option to add ground terms from quantified formulas to the term index used for CCFV (for both conflict and trigger based instantiations), which was very helpful sometimes, but I don't remember how it was in general.

Reply:
Yes, it is a bad idea. There is no need in retesting this. 
Let's focus on making the code do the right thing.